dbt/adapters/redshift/impl.py

```diff
@@ -114,6 +115,53 @@ def valid_incremental_strategies(self):
     def timestamp_add_sql(self, add_to: str, number: int = 1, interval: str = "hour") -> str:
         return f"{add_to} + interval '{number} {interval}'"
 
+    def _parse_relation_results(
+        self, database: str, schema: str, name: str, quote_policy: dict[str, bool], type: str
+    ):
+        try:
+            relation_type = self.Relation.get_relation_type(type.lower())
+        except ValueError:
+            relation_type = self.Relation.External
+        relation = self.Relation.create(
+            database=database,
+            schema=schema,
+            identifier=name,
+            quote_policy=quote_policy,
+            type=relation_type,
+        )
+        return relation
+
+    def _get_cursor(self):
+        return self.connections.get_thread_connection().handle.cursor()
+
+    def _get_tables(self, database: Optional[str], schema: Optional[str]):
+        cursor = self._get_cursor()
+        results = []
+        for table in cursor.get_tables(
+            catalog=database,
+            schema_pattern=schema,
+            table_name_pattern=None,
+            types=["VIEW", "TABLE"],
+        ):
+            results.append(
+                {"database": table[0], "schema": table[1], "name": table[2], "type": table[3]}
+            )
+            # kinda feel like it might not be a good thing to do this by index
+        return results
+
+    def list_relations_without_caching(  # type: ignore
```

Comment:
I think we can just add `@available` to make this available as a macro

Reply:
Do we want to make this available as a user-callable macro though? Its currently only being used as a macro that supports a user-callable macro such as `get_relation`. I don't believe its available to be called directly by the end user. We could certainly make this change but just double checking that is something we want to do

Reply:
I believe these are already available as a macro, after some offline conversations I don't think it being a callable macro should break anything.